File: src/main/java/org/codinjutsu/tools/nosql/commons/view/add/TextFieldWrapper.java
package org.codinjutsu.tools.nosql.commons.view.add;

import org.codinjutsu.tools.nosql.commons.model.internal.layer.DatabaseElement;

import javax.swing.JComponent;

public abstract class TextFieldWrapper<T extends JComponent> {

    protected final T component;

    TextFieldWrapper(T component) {
        this.component = component;
    }

    public abstract DatabaseElement getValue();

    public abstract void reset();

    public boolean isValueSet() {
        return true;
    }

    public JComponent getComponent() {
        return component;
    }

    public void validate() {
        if (!isValueSet()) {
            throw new IllegalArgumentException("Value is not set");
        }
    }
}

File: src/main/java/org/codinjutsu/tools/nosql/commons/view/add/TextFieldWrapper.kt
package org.codinjutsu.tools.nosql.commons.view.add

import org.codinjutsu.tools.nosql.commons.model.internal.layer.DatabaseElement
import javax.swing.JComponent

internal interface TextFieldWrapper<out T : JComponent> {
    val component: T
    val isValueSet: Boolean

    fun getValue(): DatabaseElement?
    fun reset()
    fun validate()
}